operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/DependentResourceManager.java

```diff
@@ -72,7 +73,10 @@ public DependentResourceManager(Controller<P> controller) {
   @Override
   public DeleteControl cleanup(P resource, Context context) {
     initContextIfNeeded(resource, context);
-    dependents.forEach(dependent -> dependent.delete(resource, context));
+    dependents.stream()
+        .filter(dependent -> dependent instanceof Deleter)
```

Comment:
As mentioned above, would rather put back the delete into `DependentResource`.

Reply:
I agree, there is some inconsistencies. Either Manager deal with all traits abstraction, or it is deal within abstractDependentResource  base class implementation. The former allow for more extensibility and reuse, the latter allow for simpler standalone use case.

Reply:
Done. `Deleter` is now managed in a way that consistent with the rest of the traits.